### PR TITLE
This was my add stars branch

### PR DESCRIPTION
**Pull request by itsme-kristin**
As I mentioned this morning, my add stars branch turned into a pretty big pull request. I have added the stars, changed the state for the app by clicking on a product card (switched to on image and on content because otherwise the star button would ALSO change state in addition to showing the comparison modal, which is problematic), and I have a basic, working comparison modal! It's missing the feature to show a check for true, and it could use some style, but it works! Now, I promise I'm switching to a new branch...

### CODE REVIEW

File: client/src/components/RelatedProducts/ProductCard.jsx [hunk 1]
```diff
@@ -4,72 +4,156 @@ import { makeStyles } from '@material-ui/core/styles';
...
+      var currentFeatureObject = productCardInfo.features[j];
+      if (featureData[currentFeatureObject.feature] !== undefined) {
+        if (currentFeatureObject.value === null) {
+          featureData[currentFeatureObject.feature] = 'yes!';
+        } else {
+          featureData[currentFeatureObject.feature][1] = currentFeatureObject.value;
+        }
+      } else {
+        var currentFeatureValue = [];
+        currentFeatureValue.push('');
```
Comment (bkern98) [inline client/src/components/RelatedProducts/ProductCard.jsx:99]: Is there a reason why you're pushing an empty string?

File: client/src/components/YourOutfit/OutfitCard.jsx [hunk 2]
```diff
@@ -26,11 +28,16 @@ const useStyles = makeStyles((theme) => ({
...
+  const [ratingsInfo, setRatingsInfo] = useState({
+    avgProductRating: 0,
+    totalRatings: 0
+    });
   const classes = useStyles();
 
   const getPhoto = () => {
     axios.get(`/api/products/${props.product.id}/styles`)
       .then(stylesInfo => {
+        console.log(stylesInfo.data.results[0]);
```
Comment (bkern98) [inline client/src/components/YourOutfit/OutfitCard.jsx]: Remember to remove the console.logs

File: client/src/components/YourOutfit/OutfitCard.jsx [hunk 3]
```diff
@@ -5,6 +5,8 @@ import Card from '@material-ui/core/Card';
 import CardMedia from '@material-ui/core/CardMedia';
 import CardContent from '@material-ui/core/CardContent';
 import Icon from '@material-ui/core/Icon';
+import Rating from '@material-ui/lab/Rating';
```
Comment (bkern98) [inline client/src/components/YourOutfit/OutfitCard.jsx]: No need to import Rating

File: client/src/components/RelatedProducts/ProductCard.jsx [hunk 4]
```diff
@@ -4,72 +4,156 @@ import { makeStyles } from '@material-ui/core/styles';
...
+    .catch(err => {
+      console.info('There was an error retrieving product information from the server.');
+    });
+  }
+
+  const getFeatureData = () => {
+    const featureData = {};
+
+    for (var i = 0; i < currentProductInfo.features.length; i++) {
+      var currentFeatureObject = currentProductInfo.features[i];
```
Comment (bkern98) [inline client/src/components/RelatedProducts/ProductCard.jsx]: I personally don't like the mix of es5 with es6 (using var when everywhere else is using const or let) but not a big deal

File: client/src/components/RelatedProducts/ComparisonModal.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    width: 600,
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    backgroundColor: 'white',
+    padding: '10px'
+  },
+});
+
+const ComparisonModal = React.forwardRef((props, ref) => {
```
Comment (bkern98) [inline client/src/components/RelatedProducts/ComparisonModal.jsx:28]: I'm excited to hear what forwardRef does later!

**Review by bkern98 — COMMENTED** [review]
Looks good for the most part. I added comments on some lines for some minor things to change.

File: client/src/components/RelatedProducts/ProductCard.jsx (hunk 1 above)
Comment (itsme-kristin) [reply client/src/components/RelatedProducts/ProductCard.jsx:99]: Yes! When I am populating my comparison modal, I need a placeholder for that column.

File: client/src/components/YourOutfit/OutfitCard.jsx (hunk 2 above)
Comment (itsme-kristin) [reply client/src/components/YourOutfit/OutfitCard.jsx]: Got it!

File: client/src/components/YourOutfit/OutfitCard.jsx (hunk 3 above)
Comment (itsme-kristin) [reply client/src/components/YourOutfit/OutfitCard.jsx]: Fixed it!

File: client/src/components/RelatedProducts/ProductCard.jsx (hunk 4 above)
Comment (itsme-kristin) [reply client/src/components/RelatedProducts/ProductCard.jsx]: I think I could probably get away with let. I am using the same variable name for different circumstances, so I thought const would be problematic.

File: client/src/components/RelatedProducts/ComparisonModal.jsx (hunk 5 above)
Comment (itsme-kristin) [reply client/src/components/RelatedProducts/ComparisonModal.jsx:28]: Still figuring it out! But I DO know that there is an error message if you don't use it. : )

**Review by bkern98 — APPROVED** [review]
Good to go!

**Pull request merged**